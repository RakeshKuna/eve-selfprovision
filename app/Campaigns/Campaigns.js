/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { ETable } from 'containers/ETable';
import { Link } from 'react-router-dom';
import { Button, Row, Col } from 'reactstrap';
import { history } from 'helpers';
import userService from 'services';
import { alertActions } from 'actions';
import { store } from '../store';


export class Campaigns extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      loading: false
    };
    this.columns = [{
      Header: 'Product',
      accessor: 'product' // String-based value accessors!
    }, {
      Header: 'DMA Name',
      accessor: 'dmaName'
    }, {
      Header: 'Start Date',
      accessor: 'startDate'
    }, {
      Header: 'Promotion Type',
      accessor: 'promotionType' // String-based value accessors!
    }, {
      Header: 'Promotion Name',
      accessor: 'promotionName'
    }, {
      Header: 'Promotional Price',
      accessor: 'promotionalPrice'
    }, {
      Header: 'Display Name',
      accessor: 'displayName'
    }, {
      Header: 'Promotion ID',
      accessor: 'promotionID'
    }, {
      Header: 'SKU OR QuickCode',
      accessor: 'skuORQuickCode'
    }, {
      Header: 'Actions',
      accessor: '[row identifier to be passed to button]',
      Cell: (row) => (<Button color="primary" onClick={() => this.editCampaigns(row.original)} className="view_btn">Edit</Button>),
      filterable: false
    }];
  }


     getCampaigns=(skus) => {
       this.setState({
         loading: true
       });
       userService.getPromotionsList(this.props.channelPartnerId, skus)
         .then(
           (success) => {
             if (!Array.isArray(success.promotionsResponseMessage)) {
               success.promotionsResponseMessage = [success.promotionsResponseMessage];
             }
             success.promotionsResponseMessage.forEach((result) => {
               if (parseInt(result.startDate, 10)) {
                 const startDate = new Date(parseInt(result.startDate, 10));
                 result.startDate = `${startDate.getMonth() + 1}/${startDate.getDate()}/${startDate.getFullYear()}`;
               }
               // result.startDate = new Date(parseInt(result.startDate)).toLocaleDateString();
             });
             this.setState({
               data: success.promotionsResponseMessage,
               loading: false
             });
           },
           (error) => {
             store.dispatch(alertActions.error(error));
             this.setState({
               loading: false
             });
           });
     }
     editCampaigns=(campaign) => {
       history.push({ pathname: `/${this.props.channelPartnerId}/updateCampaign/${campaign.promotionName}`, state: { campaign } });
     }
     render() {
       return (
         <div>
           <Row className="mb-4">
             <Col sm={8}>
               <h2>Campaigns</h2>
             </Col>
             <Col sm={4} className="text-right">
               <Button color="primary" className="mr-2" tag={Link} to={`/${this.props.channelPartnerId}/addCampaign`}>ADD</Button>
             </Col>
           </Row>
           <Row>
             <Col>
               <ETable key="A" data={this.state.data} columns={this.columns} loading={this.state.loading} />
             </Col>
           </Row>
         </div>
       );
     }
}
export default Campaigns;
