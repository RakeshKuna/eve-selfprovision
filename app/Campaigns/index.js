export { default as Campaigns } from './Campaigns';
export { default as AddCampaigns } from './AddCampaigns';
