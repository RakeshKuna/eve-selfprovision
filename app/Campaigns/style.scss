@import 'styles/global-styles.scss';
@import 'styles/common.scss';
.datepicker {
    .react-datepicker-wrapper,
    .react-datepicker__input-container {
        display: block;
    }
}

.check_main {
    &.form-check-inline {
        .form-check-label {
            margin-right: 10px;
        }
    }
    @media screen and (max-width: $screen-xxs-max) {
        &.form-check-inline {
            display: block;
            .form-check-label {
                display: block;
            }
        }
    }
}