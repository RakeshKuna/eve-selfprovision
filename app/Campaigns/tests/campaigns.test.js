import React from 'react';
import { shallow } from 'enzyme';
import { Campaigns } from '../Campaigns';

describe('Campaigns', () => {
  it('<Row> element should render', () => {
    const wrapper = shallow(<Campaigns />);
    expect(wrapper).toMatchSnapshot();
  });
  it('<Row> on Button redirect', () => {
    const wrapper = shallow(<Campaigns />);
    wrapper.find('Button').at(0).props('to');
  });
});
