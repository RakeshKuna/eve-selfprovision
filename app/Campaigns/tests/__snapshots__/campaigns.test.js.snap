// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaigns <Row> element should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Campaigns />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          className="mb-4"
          tag="div"
        >
          <Col
            sm={8}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h2>
              Campaigns
            </h2>
          </Col>
          <Col
            className="text-right"
            sm={4}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              className="mr-2"
              color="primary"
              tag={[Function]}
              to="/undefined/addCampaign"
            >
              ADD
            </Button>
          </Col>
        </Row>,
        <Row
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Product",
                    "accessor": "product",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Start Date",
                    "accessor": "startDate",
                  },
                  Object {
                    "Header": "Promotion Type",
                    "accessor": "promotionType",
                  },
                  Object {
                    "Header": "Promotion Name",
                    "accessor": "promotionName",
                  },
                  Object {
                    "Header": "Promotional Price",
                    "accessor": "promotionalPrice",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Promotion ID",
                    "accessor": "promotionID",
                  },
                  Object {
                    "Header": "SKU OR QuickCode",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />
          </Col>
        </Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Campaigns
              </h2>
            </Col>,
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/undefined/addCampaign"
              >
                ADD
              </Button>
            </Col>,
          ],
          "className": "mb-4",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h2>
                Campaigns
              </h2>,
              "sm": 8,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Campaigns",
              },
              "ref": null,
              "rendered": "Campaigns",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/undefined/addCampaign"
              >
                ADD
              </Button>,
              "className": "text-right",
              "sm": 4,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "ADD",
                "className": "mr-2",
                "color": "primary",
                "tag": [Function],
                "to": "/undefined/addCampaign",
              },
              "ref": null,
              "rendered": "ADD",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Product",
                    "accessor": "product",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Start Date",
                    "accessor": "startDate",
                  },
                  Object {
                    "Header": "Promotion Type",
                    "accessor": "promotionType",
                  },
                  Object {
                    "Header": "Promotion Name",
                    "accessor": "promotionName",
                  },
                  Object {
                    "Header": "Promotional Price",
                    "accessor": "promotionalPrice",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Promotion ID",
                    "accessor": "promotionID",
                  },
                  Object {
                    "Header": "SKU OR QuickCode",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Product",
                    "accessor": "product",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Start Date",
                    "accessor": "startDate",
                  },
                  Object {
                    "Header": "Promotion Type",
                    "accessor": "promotionType",
                  },
                  Object {
                    "Header": "Promotion Name",
                    "accessor": "promotionName",
                  },
                  Object {
                    "Header": "Promotional Price",
                    "accessor": "promotionalPrice",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Promotion ID",
                    "accessor": "promotionID",
                  },
                  Object {
                    "Header": "SKU OR QuickCode",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "A",
            "nodeType": "class",
            "props": Object {
              "columns": Array [
                Object {
                  "Header": "Product",
                  "accessor": "product",
                },
                Object {
                  "Header": "DMA Name",
                  "accessor": "dmaName",
                },
                Object {
                  "Header": "Start Date",
                  "accessor": "startDate",
                },
                Object {
                  "Header": "Promotion Type",
                  "accessor": "promotionType",
                },
                Object {
                  "Header": "Promotion Name",
                  "accessor": "promotionName",
                },
                Object {
                  "Header": "Promotional Price",
                  "accessor": "promotionalPrice",
                },
                Object {
                  "Header": "Display Name",
                  "accessor": "displayName",
                },
                Object {
                  "Header": "Promotion ID",
                  "accessor": "promotionID",
                },
                Object {
                  "Header": "SKU OR QuickCode",
                  "accessor": "skuORQuickCode",
                },
                Object {
                  "Cell": [Function],
                  "Header": "Actions",
                  "accessor": "[row identifier to be passed to button]",
                  "filterable": false,
                },
              ],
              "data": Array [],
              "loading": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            className="mb-4"
            tag="div"
          >
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Campaigns
              </h2>
            </Col>
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/undefined/addCampaign"
              >
                ADD
              </Button>
            </Col>
          </Row>,
          <Row
            tag="div"
          >
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Product",
                      "accessor": "product",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Start Date",
                      "accessor": "startDate",
                    },
                    Object {
                      "Header": "Promotion Type",
                      "accessor": "promotionType",
                    },
                    Object {
                      "Header": "Promotion Name",
                      "accessor": "promotionName",
                    },
                    Object {
                      "Header": "Promotional Price",
                      "accessor": "promotionalPrice",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Promotion ID",
                      "accessor": "promotionID",
                    },
                    Object {
                      "Header": "SKU OR QuickCode",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />
            </Col>
          </Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                sm={8}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h2>
                  Campaigns
                </h2>
              </Col>,
              <Col
                className="text-right"
                sm={4}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Button
                  className="mr-2"
                  color="primary"
                  tag={[Function]}
                  to="/undefined/addCampaign"
                >
                  ADD
                </Button>
              </Col>,
            ],
            "className": "mb-4",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h2>
                  Campaigns
                </h2>,
                "sm": 8,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Campaigns",
                },
                "ref": null,
                "rendered": "Campaigns",
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  className="mr-2"
                  color="primary"
                  tag={[Function]}
                  to="/undefined/addCampaign"
                >
                  ADD
                </Button>,
                "className": "text-right",
                "sm": 4,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "ADD",
                  "className": "mr-2",
                  "color": "primary",
                  "tag": [Function],
                  "to": "/undefined/addCampaign",
                },
                "ref": null,
                "rendered": "ADD",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Product",
                      "accessor": "product",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Start Date",
                      "accessor": "startDate",
                    },
                    Object {
                      "Header": "Promotion Type",
                      "accessor": "promotionType",
                    },
                    Object {
                      "Header": "Promotion Name",
                      "accessor": "promotionName",
                    },
                    Object {
                      "Header": "Promotional Price",
                      "accessor": "promotionalPrice",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Promotion ID",
                      "accessor": "promotionID",
                    },
                    Object {
                      "Header": "SKU OR QuickCode",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Product",
                      "accessor": "product",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Start Date",
                      "accessor": "startDate",
                    },
                    Object {
                      "Header": "Promotion Type",
                      "accessor": "promotionType",
                    },
                    Object {
                      "Header": "Promotion Name",
                      "accessor": "promotionName",
                    },
                    Object {
                      "Header": "Promotional Price",
                      "accessor": "promotionalPrice",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Promotion ID",
                      "accessor": "promotionID",
                    },
                    Object {
                      "Header": "SKU OR QuickCode",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "A",
              "nodeType": "class",
              "props": Object {
                "columns": Array [
                  Object {
                    "Header": "Product",
                    "accessor": "product",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Start Date",
                    "accessor": "startDate",
                  },
                  Object {
                    "Header": "Promotion Type",
                    "accessor": "promotionType",
                  },
                  Object {
                    "Header": "Promotion Name",
                    "accessor": "promotionName",
                  },
                  Object {
                    "Header": "Promotional Price",
                    "accessor": "promotionalPrice",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Promotion ID",
                    "accessor": "promotionID",
                  },
                  Object {
                    "Header": "SKU OR QuickCode",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ],
                "data": Array [],
                "loading": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
